src/main/java/com/scottlogic/filters/KeywordContentPostFilter.java

```diff
@@ -21,7 +22,7 @@ public KeywordContentPostFilter(String keywordInput) {
         }
         if(Objects.equals(Keyword, "")){return new ArrayList<>();}
         List<UserPost> outputList = new ArrayList<>(inputList);
-        outputList.removeIf(a -> !a.getContents().contains(Keyword));
+        outputList.removeIf(a -> !a.getContents().toLowerCase().contains(Keyword.toLowerCase()));
```

Comment:
At the moment whenever you call this function you are doing `Keyword.toLowerCase()` as you know that the `Keyword` can't be used in any other way doing this change in the constructor might make the code cleaner and more efficient

Reply:
What would happen if `null` is fed in as the `Keyword` and then you end up called `.toLowerCase()` on it?